### PR TITLE
Refactor and enable workspace startup reporter

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
Refactors and enables the WorkspaceStartupReporter for our [jenkins job](https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/OSIO_CI/job/workspace-startup-reporter/).

### What issues does this PR fix or reference?
Follow-up of https://github.com/redhat-developer/che-functional-tests/issues/548 as part of migrating our jobs scripts to rh-che repository.

### How have you tested this PR?
manual testing locally

### Extra

I have a concern with the current state of the reporter application.
It works as intended, but my concern is the limits where the applications trigger warnings.
When there is a long stable period within the data and a sudden spike appears, a change in the startup time of 0.5 seconds can have as much impact as 10% in diff.

My question is whether we should re-work the coloration of the attachment and switch them from percentage trigger to a static amount of seconds needed in change to trigger a warning or error.

Here is a graphic representation of what I mean:
Screenshot_20190924_133159

e.g.: Cluster PREVIEW_2A stop has a diff of 0.46s in PVC mode and 1.25s in Ephemeral mode and they contributed as much as 9% and 16% respectively which would trigger a failure.

Currently the values are set:
* anything below 1% deviation is green
* (1;5)% is orange - acceptable range
* anything above 5% is red

### CODE REVIEW

File: .ci/workspace-startup/start-workspace-reporter/src/main/java/com/redhat/che/start_workspace_reporter/ReporterMain.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package com.redhat.che.start_workspace_reporter;
```
[the remark this answers is not in the document]
Comment (ScrewTSW) [reply .ci/workspace-startup/start-workspace-reporter/src/main/java/com/redhat/che/start_workspace_reporter/ReporterMain.java]: good point, I always forget about those

**Comment by ScrewTSW** [comment]
> Generally LGTM, though java license headers are missing.
> 
> I'm not entirely sure keeping these tests in the `.ci` directory is the best option; does it make sense to rework the repo dir structure so that there's some tests directory that stores this test alongside `functional-tests`, `load-tests`, etc.? This could be a separate issue, regardless.
> 
> Also I agree with your concern on percent difference not being a great metric when the overall duration is small (0.5s is 10% of 5 seconds, but we don't really care about that). Perhaps it makes sense to add some logic to the `getColorBasedOnPercentage` method, where it also takes into account the absolute size of the difference (<2s green, <5s yellow, >5s decided based off percent change as current?). There's probably some fancier math we could apply here but it's maybe overkill.

I will be absolutely happy to move it over. Wasn't sure where was a good place for it, since it's being only run as part of CI which is obscurely hidden behind a VPN anyways. It also relies on other architecture like Zabbix server and other tests that provide all the experimental data.

I am happy to implement a logic for seconds alongside percent change in the method. I think it's a really good idea.

File: .ci/workspace-startup/start-workspace-reporter/src/main/java/com/redhat/che/start_workspace_reporter/ReporterMain.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibuziuk) [reply .ci/workspace-startup/start-workspace-reporter/src/main/java/com/redhat/che/start_workspace_reporter/ReporterMain.java]: please fix

**Comment by ScrewTSW** [comment]
> @ScrewTSW are there any docs planned for start_workspace_reporter app?

Yes. I believe I have a separate documentation issue.

**Comment by ibuziuk** [comment]
> Yes. I believe I have a separate documentation issue.

link ?

File: .ci/workspace-startup/start-workspace-reporter/src/main/java/com/redhat/che/start_workspace_reporter/ReporterMain.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScrewTSW) [reply .ci/workspace-startup/start-workspace-reporter/src/main/java/com/redhat/che/start_workspace_reporter/ReporterMain.java]: done.

**Comment by ScrewTSW** [comment]
> > Yes. I believe I have a separate documentation issue.
> 
> link ?

https://github.com/redhat-developer/rh-che/issues/1588
This issue encompasses WorkspaceStartupReporter as well

**Comment by ScrewTSW** [comment]
> LGTM, but please add the docs with general info and instructions how to build / run the app

will do.

**Pull request merged**